tsl/src/bgw_policy/job.c

```diff
@@ -504,6 +505,19 @@ job_execute(BgwJob *job)
 							InvalidOid,
 							COERCE_EXPLICIT_CALL);
 
+	/* Here we create a query string from the function/procedure name that we
+	 * are calling. We do not update the status after the execution has
+	 * finished since this is wrapped inside the code that starts and stops
+	 * any job, not just custom jobs. We just provide more detailed
+	 * information here that we are actually calling a specific custom
+	 * function. */
+	query = makeStringInfo();
+	appendStringInfo(query,
+					 "CALL %s.%s()",
+					 NameStr(job->fd.proc_schema),
+					 NameStr(job->fd.proc_name));
+	pgstat_report_activity(STATE_RUNNING, query->data);
```

Comment:
this needs a test

Reply:
Added one. I actually wanted to add a more elaborate test that shows what statement is being executed inside a custom procedure, but that proved to be difficult.